blockwork/build/transform.py

```diff
@@ -130,7 +159,31 @@ def bind_inputs(self, **interface: Interface):
         may be supplied. Resolved values will be passed through to the execute
         method accordingly as a single value or array of values.
         """
-        return self._bind_interfaces(Direction.INPUT, Pipe.FLOW, **interface)
+        self._bind_interfaces(Direction.INPUT, Pipe.FLOW, **interface)
+        return self
+    
+    def bind_tools(self, *tools: type["Tool"]) -> Self:
+        """
+        Attach additional tools to the instance - typically only used if 
+        a transform is used anonymously.
+        """
+        self._tools += tools
+        return self
+    
+    def bind_execute(self, 
+                     execute: Execute) -> Self:
+        """
+        Attach an alternative execute method to the instance - typically only
+        used if a transform is used anonymously.
+        """
+        self.execute = execute
+        return self
+    
+    def get_hashsource(self, ctx: "Context") -> str:
```

Comment:
It might be worth a comment just above these saying that the decorator isn't used, but that the init method applies caching